lib/ActivityStreams.js

```diff
@@ -443,22 +456,31 @@ ActivityStreams.prototype = {
    * Unload the application
    */
   unload(reason) { // eslint-disable-line no-unused-vars
+    let defaultUnload = () => {
+      clearTimeout(this._previewCacheTimeoutID);
+      this._previewProvider.uninit();
+      NewTabURL.reset();
+      this.workers.clear();
+      this._removeListeners();
+      this._pagemod.destroy();
+      this._button.destroy();
+      this._tabTracker.uninit();
+      this._telemetrySender.uninit();
+      this._appURLHider.uninit();
+      this._perfMeter.uninit();
+      this._memoizer.uninit();
+      this._populatingCache.places = false;
+      this._populatingCache.preview = false;
+    };
+
     switch (reason){
       // can be one of: uninstall/disable/shutdown/upgrade/downgrade
+      case "uninstall":
```

Comment:
specifically comment 12 https://bugzilla.mozilla.org/show_bug.cgi?id=627432#c12

You might want to do `disable` instead?